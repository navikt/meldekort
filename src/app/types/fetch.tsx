import axios, { AxiosRequestTransformer } from 'axios';

const dateTransformer: AxiosRequestTransformer = data => {
  if (data instanceof Date) {
    // YYYY-MM-DD
    return (
      data.getFullYear() +
      '-' +
      (data.getMonth() + 1).toString().padStart(2, '0') +
      '-' +
      data.getDate()
    );
  }
  if (Array.isArray(data)) {
    return data.map(val => dateTransformer(val));
  }
  if (typeof data === 'object' && data !== null) {
    return Object.fromEntries(
      Object.entries(data).map(([key, val]) => [key, dateTransformer(val)])
    );
  }
  return data;
};

export const prefferedAxios = axios.create({
  transformRequest: axios.defaults.transformRequest
    ? [dateTransformer].concat(axios.defaults.transformRequest)
    : dateTransformer,
  baseURL: window.location.origin,
});

export enum FetchStatus {
  UNFETCHED,
  IN_PROGRESS,
  SUCCESS,
  FAILURE,
}

export function hentDataFraKilde<T>(
  fetchState: FetchState<T>,
  emptyVerdi: any
): T {
  return fetchState && fetchState.status === FetchStatus.SUCCESS
    ? fetchState.data
    : emptyVerdi;
}

export type FetchState<T> =
  | {
      status: FetchStatus.UNFETCHED;
    }
  | {
      status: FetchStatus.IN_PROGRESS;
    }
  | {
      status: FetchStatus.SUCCESS;
      data: T;
    }
  | {
      status: FetchStatus.FAILURE;
      error: unknown;
    };
