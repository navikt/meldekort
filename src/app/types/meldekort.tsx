// CONSTANTS / Action?
export enum Constants {
    LEGG_TIL_AKTIVT_MELDEKORT = 'LEGG_TIL_AKTIVT_MELDEKORT'
}

/* INTERFACES */
export interface Meldekort {
    meldekortId: number;
    kortType: KortType;
    meldeperiode: Meldeperiode;
    meldegruppe: Meldegruppe;
    kortStatus: KortStatus;
    bruttoBelop: number;
    mottattDato: Date;
    korrigerbart: boolean;
}

// hentMeldekortDetaljer
export interface Meldekortdetaljer {
    id: string;
    personId: number;
    fodselsnr: string;
    meldekortId: number;
    meldeperiode: string;
    arkivnokkel: string;
    kortType: KortType;
    meldeDato: Date;
    lestDato: Date;
    sporsmal: Sporsmal;
    begrunnelse: string;
}

export interface Meldeperiode {
    fra: Date;
    til: Date;
    kortKanSendesFra: Date;
    periodeKode: string;
}

export interface MeldekortdetaljerInnsending {
    meldekortId: number;
    kortType: KortType;
    meldegruppe: Meldegruppe;
    mottattDato: Date;
    meldeperiode: Meldeperiode;
    erArbeidssokerNestePeriode: boolean;
    bruttoBelop: number;
    fravaersdager: Fravaer[];
    korrigerbart: boolean;
    begrunnelse: string;

    fnr: string;
    personId: number;
    ipAdresse: string;
    sessjonsId: string;
}

export interface Fravaer {
    dag: Date;
    type: FravaerType;
    arbeidTimer: number;
}

export interface ValideringsResultat {
    meldekortId: number;
    status: string;
    arsakskoder: Arsakskode[];
    meldekortdager: MeldekortDag[];
}

export interface Arsakskode {
    kode: string;
    tekst: string;
}

export interface Sporsmal {
    arbeidssoker: boolean;
    arbeidet: boolean;
    syk: boolean;
    annetFravaer: boolean;
    kurs: boolean;
    signatur: boolean;
    meldekortDager: MeldekortDag[];
}

export interface MeldekortDag {
    dag: number;
    arbeidetTimerSum: number;
    syk: boolean;
    annetFravaer: boolean;
    kurs: boolean;
    meldegruppe: string;
}

/* ENUMS */
export enum KortType {
    ORDINAER = '01',
    ERSTATNING = '03',
    RETUR = '04',
    ELEKTRONISK = '05',
    AAP = '06',
    ORDINAER_MANUELL = '07',
    MASKINELT_OPPDATERT = '08',
    MANUELL_ARENA = '09',
    KORRIGERT_ELEKTRONISK = '10'
}

export enum Meldegruppe {
    ATTF,
    DAGP,
    INDIV,
    ARBS,
    FY,
    NULL
}

export enum KortStatus {
    OPPRE = 'OPPRE',
    SENDT = 'SENDT',
    SLETT = 'SLETT',
    REGIS = 'REGIS' ,
    FMOPP = 'FMOPP',
    FUOPP = 'FUOPP',
    KLAR = 'KLAR',
    KAND = 'KAND',
    IKKE = 'IKKE',
    OVERM = 'OVERM',
    NYKTR = 'NYKTR',
    FERDI = 'FERDI',
    FEIL = 'FEIL',
    VENTE = 'VENTE',
    OPPF = 'OPPF'
}

export enum FravaerType {
    KURS_UTDANNING = 'K',
    SYKDOM = 'S',
    ANNET_FRAVAER = 'X',
    ARBEIDS_FRAVAER = 'A'
}

// Internt bruk

export interface SporsmalOgSvar {
    sporsmalId: string;
    svar: boolean;
    formatertDato?: string;
}
