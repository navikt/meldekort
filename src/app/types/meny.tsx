import { MenyPunkt } from '../utils/menyConfig';

export interface MenyState {
  valgtMenyPunkt: MenyPunkt;
  alleMenyPunkter: MenyPunkt[];
  erApen: boolean;
}

export enum MenyTypeKeys {
  SETT_VALGT_MENYPUNKT = 'SETT_VALGT_MENYPUNKT',
  SETT_AKTIVE_MENYPUNKTER = 'SETT_AKTIVE_MENYPUNKTER',
  TOGGLE_MENY = 'TOGGLE_MENY',
}
