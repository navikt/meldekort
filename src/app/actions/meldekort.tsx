import { ActionType, createStandardAction } from 'typesafe-actions';
import { AxiosError } from 'axios';
import { SendtMeldekort } from '../types/meldekort';

export enum MeldekortTypeKeys {
    API_KALL_FEILET = 'API_KALL_FEILET',
    LEGG_TIL_INNSENDT_MELDEKORT = 'LEGG_TIL_INNSENDT_MELDEKORT'
}

export const MeldekortActions = {
    apiKallFeilet: createStandardAction(MeldekortTypeKeys.API_KALL_FEILET)<AxiosError>(),
    leggTilInnsendtMeldekort: createStandardAction(MeldekortTypeKeys.LEGG_TIL_INNSENDT_MELDEKORT)<SendtMeldekort[]>()
};

export type MeldekortActionTypes = ActionType<typeof MeldekortActions>;