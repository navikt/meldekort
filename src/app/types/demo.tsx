// Defining the redux state and constants
import * as actions from '../actions/demo';
import { ActionType } from 'typesafe-actions';

// import { DemoActions } when you need to use our actions
export type DemoActions = ActionType<typeof actions>;

// STATE
// A simple one field 'list' to store list of string in our state
export interface DemoState {
    list: string[];
    loading: boolean;
}

// CONSTANTS / Action?
export enum Constants {
    ADD_ITEM = 'ADD_ITEM',
    SET_LOADING = 'SET_LOADING'
}