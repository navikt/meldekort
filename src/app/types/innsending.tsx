import { KortStatus, Sporsmal } from './meldekort';

export enum Constants {
    LEGG_TIL_SVAR = 'LEGG_TIL_SVAR',
    SETT_MELDEKORTINFO = "SETT_MELDEKORTINTO",
}

export interface Innsending {
    meldekortId: number;
    kortStatus: KortStatus;
    sporsmal: Sporsmal;
    // Sett inn vars for utfylling.
}