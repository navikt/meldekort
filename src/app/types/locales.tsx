import * as actions from '../actions/locales';
import { ActionType } from 'typesafe-actions';

export type LocalesActions = ActionType<typeof actions>;

export enum Constants {
  UPDATE_LOCALES = 'UPDATE_LOCALES',
}

export enum LocaleEnum {
  en = 'en',
  nb = 'nb',
}
