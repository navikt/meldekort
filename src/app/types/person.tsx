import { Meldekort } from './meldekort';

// fetchMeldekort
export interface Person {
  maalformkode: string;
  meldeform: MeldeForm;
  meldekort: Meldekort[];
  etterregistrerteMeldekort: Meldekort[];
  fravaer: Fravaer[];
  id: string;
  antallGjenstaaendeFeriedager: number;
}

export interface PersonInfo {
  personId: number;
  fodselsnr: string;
  etternavn: string;
  fornavn: string;
}

// hentPersonstatus
export interface PersonStatus {
  id: string;
  statusArbeidsoker: string;
  statusYtelse: string;
}

export interface MeldeformDetaljerInn {
  meldeformNavn: string;
}

export interface Meldeperiode {
  id: string;
  meldeperiodeNavn: string;
  fraDato: Date;
  tilDato: Date;
}

export enum MeldeForm {
  ELEKTRONISK = 'EMELD',
  PAPIR = 'PAPIR',
  MANUELL = 'MANU',
  AUTO = 'AUTO',
  IKKE_SATT = 'IKKE SATT',
}

export interface Fravaer {
  fraDato: Date;
  tilDato: Date;
  type: FravaerType;
}

export enum FravaerType {
  KURS_UTDANNING = 'K',
  SYKDOM = 'S',
  ANNET_FRAVAER = 'X',
  ARBEIDS_FRAVAER = 'F',
}
