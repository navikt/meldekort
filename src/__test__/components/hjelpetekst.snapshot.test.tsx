import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ProviderWrapper from '../utils/providerWrapper';
import Hjelpetekst from '../../app/components/meldekortdetaljer/ukevisning/hjelpetekst';
import { MeldekortDag } from '../../app/types/meldekort';

it('Sprakvelger', () => {
  const tree = renderer.create(
    <ProviderWrapper>
      <Hjelpetekst meldekortDag={mkDag} erAap={true} />
    </ProviderWrapper>
  );
  expect(tree).toMatchSnapshot();
});

const mkDag: MeldekortDag = {
  dag: 1,
  arbeidetTimerSum: 7.5,
  syk: false,
  annetFravaer: true,
  kurs: false,
};
