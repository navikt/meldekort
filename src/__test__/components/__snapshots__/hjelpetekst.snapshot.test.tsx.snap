// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hjelpetekst 1`] = `
<span
  className="meldekortdetaljer-utfyllt-hjelpetekster"
>
  <span>
    <span
      className="overskrift-hjelpetekst"
    >
      <strong>
        ARBEID 
      </strong>
    </span>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "Du skal registrere alle timene du har hatt lønnet arbeid, også i egen næringsvirksomhet. Dette gjelder alle dager i uken, også lørdag og søndag, uansett når på døgnet du jobber.  <br> <br> Unntaket er dersom du får lønn for flere timer enn du faktisk har jobbet. Da skal du føre opp alle timene du får lønn for. <br> <br> Hvis du har en prosentstilling, må du regne den om til timer. Full stilling er vanligvis 37,5 timer i uken. <br> <br> Kontakt NAV hvis du <ul>  <li> er i tvil om hvor mange timer du skal føre opp </li>  <li> har et yrke som ikke har 37,5 timer per uke som normal arbeidstid </li>  <li> deltar på deltidstiltak eller tiltak med lønn  </li>  </ul> ",
        }
      }
    />
  </span>
  <span>
    <span
      className="overskrift-hjelpetekst"
    >
      <strong>
        FERIE OG ANNET FRAVÆR ENN SYKDOM 
      </strong>
    </span>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "<b><p style=\\"color:#ee5522;\\">  Fravær fra jobb på grunn av karantene føres på samme måte som sykdom (gjelder ikke ved hjemmekontor).  </p></b>  Du skal registrere de dagene du ikke har kunnet ta arbeid eller gjennomføre avtalt aktivitet, og dette skyldes annet enn egen sykdom. <br> <br> Det gjelder f.eks. hvis du har hatt ferie og derfor ikke kunnet gjennomføre avtalt aktivitet. <br> <br> Det gjelder også opphold i utlandet som ikke er godkjent av NAV. Er du statsborger i et EU/EØS-land og oppholder deg i et annet EU/EØS-land, skal oppholdet ikke godkjennes av NAV. <br> <br> Det gjelder også dersom du ikke ønsker å være tilgjengelig for NAV på dager uten planlagt aktivitet. <br> <br> Hvis du likevel ønsker å få arbeidsavklaringspenger for fraværsdagene, må du kontakte NAV før fraværet starter. ",
        }
      }
    />
  </span>
</span>
`;
